### PR TITLE
Demo frontend

### PR DESCRIPTION
**Pull request by mbrandenburger**
This PR adds a snapshot of the clock auction UI and small mock backend.

### CODE REVIEW

**Comment by g2flyer** [comment]
there are lots of commits and changed files here and it is WIP, so making it a bit harder to review:  The changes in ecc we certainly will have to review and the backend is also fairly manageable and given the potential for generalization also probably good to get many eyes. The front-end part, though, is rather large and hard to assess if you don't know and haven't used Vue before: is there anything to look at there? Or would it maybe make sense that in a call you give a high-level overview over the UI?

**Comment by mbrandenburger** [comment]
> there are lots of commits and changed files here and it is WIP, so making it a bit harder to review: The changes in ecc we certainly will have to review and the backend is also fairly manageable and given the potential for generalization also probably good to get many eyes. The front-end part, though, is rather large and hard to assess if you don't know and haven't used Vue before: is there anything to look at there? Or would it maybe make sense that in a call you give a high-level overview over the UI?

what I can do to make reviewer's life easier is to split this PR into two. One for the mock backend including auction go chaincode and refactoring of ecc, and another one just for the frontend.

In particular, the ecc refactoring is the only "critical" change to FPC, whereas the other components provided in this PR (i.e. the go mock chaincode, the mock backend, and the UI) are only used for the demo.

And yes, I can give you a code walk-through.

The drawback of having two PRs here that the UI cannot be easily tested as those components

**Comment by mbrandenburger** [comment]
I removed the mock chaincode from this branch, so it only contains the frontend code. 

The mock chaincode is now available in #194

File: demo/client/frontend/src/components/AppHeader.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          <v-btn color="secondary" dark v-on="on">
+            Round {{ auction.clockRound }}
+          </v-btn>
+        </template>
+        <v-list>
+          <v-list-item
+            v-for="index in 4"
+            :key="index"
+            @click="UPDATE_CLOCK_ROUND(index)"
+          >
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply demo/client/frontend/src/components/AppHeader.vue]: You are right. The final version will not have the capability to change the current round this way. For now, this is just for debugging and will be removed later. Maybe the UI debug mode I mentioned in my other comment is handy here as well.

File: demo/client/frontend/src/components/AppMenu.vue [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+      let defaultItems = [
+        {
+          title: "Auction Info",
+          link: "/auction_info",
+          icon: "fa-info-circle"
+        },
+        { title: "Auction History", link: "", icon: "fa-calendar" },
+        { title: "Settings", link: "", icon: "fa-cog" },
+        { title: "Help", link: "", icon: "fa-question-circle" }
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply demo/client/frontend/src/components/AppMenu.vue]: Right, right now there is no function. But I guess every complicated website needs a `help` and `settings` section. At the moment, the only reason why this is there, is that it looks more like a real website :)

File: demo/client/frontend/src/components/AuctionDetails.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        <v-flex lg12>
+          <v-card>
+            <v-card-text class="pa-0">
+              {{ auction }}
+            </v-card-text>
+          </v-card>
+        </v-flex>
+      </v-layout>
+    </v-container>
+  </div>
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply demo/client/frontend/src/components/AuctionDetails.vue]: yes! I propose to keep it as it is for the moment and add a debug mode later. There are other debugging functions (e.g. setting phase and round) which need to go away in normal demo mode.

[the remark this answers is not in the document]
Comment (mbrandenburger) [reply demo/client/frontend/src/components/AuctionDetails.vue]: I added a debug label to this panel and out it as pretty json, thus it does not harm your eye anymore

**Comment by mbrandenburger** [comment]
> Minor UI modifications: In Auction Info view,
> 
> 1. For completeness, display Auction Name on the top?
> 2. "CLOCK PHASE" in place of "CLOCK"?  (Similarly, "Assign Phase" in place of "Assign")

> Reg. Display of the user's role (on the left top of display):
> Display "Bidder" and "Auctioneer" instead of "auction.bidder" and "auction.auctioneer" ?

good suggestions! this is fixed now!

**Comment by mbrandenburger** [comment]
@swetharepakula Please have a look and consider this PR for merging. Thanks!

**Pull request merged**